drivers/adc/adc_stm32.c

```diff
@@ -1894,42 +1894,37 @@ DT_INST_FOREACH_STATUS_OKAY(GENERATE_ISR)
 			(ADC_DMA_CHANNEL_INIT(id, src, dest)),				\
 			(/* Required for other adc instances without dma */))
 
-#define ADC_STM32_INIT(index)						\
-ADC_STM32_CHECK_DT_CLOCK(index)                                         \
-									\
-PINCTRL_DT_INST_DEFINE(index);						\
-									\
-static const struct stm32_pclken pclken_##index[] =			\
-				 STM32_DT_INST_CLOCKS(index);		\
-									\
-static const struct adc_stm32_cfg adc_stm32_cfg_##index = {		\
-	.base = (ADC_TypeDef *)DT_INST_REG_ADDR(index),			\
-	ADC_STM32_IRQ_FUNC(index)					\
-	.pclken = pclken_##index,					\
-	.pclk_len = DT_INST_NUM_CLOCKS(index),				\
-	.clk_prescaler = ADC_STM32_DT_PRESC(index),			\
-	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),			\
-	.sequencer_type = DT_INST_PROP(index, st_adc_sequencer),	\
-	.sampling_time_table = DT_INST_PROP(index, sampling_times),	\
-	.num_sampling_time_common_channels =				\
-		DT_INST_PROP_OR(index, num_sampling_time_common_channels, 0),\
-	.res_table_size = DT_INST_PROP_LEN(index, resolutions),		\
-	.res_table = DT_INST_PROP(index, resolutions),			\
-};									\
-									\
-static struct adc_stm32_data adc_stm32_data_##index = {			\
-	ADC_CONTEXT_INIT_TIMER(adc_stm32_data_##index, ctx),		\
-	ADC_CONTEXT_INIT_LOCK(adc_stm32_data_##index, ctx),		\
-	ADC_CONTEXT_INIT_SYNC(adc_stm32_data_##index, ctx),		\
-	ADC_DMA_CHANNEL(index, PERIPHERAL, MEMORY)			\
-};									\
-									\
-PM_DEVICE_DT_INST_DEFINE(index, adc_stm32_pm_action);			\
-									\
-DEVICE_DT_INST_DEFINE(index,						\
-		    &adc_stm32_init, PM_DEVICE_DT_INST_GET(index),	\
-		    &adc_stm32_data_##index, &adc_stm32_cfg_##index,	\
-		    POST_KERNEL, CONFIG_ADC_INIT_PRIORITY,		\
-		    &api_stm32_driver_api);
+#define ADC_STM32_INIT(index)                                                                      \
+	ADC_STM32_CHECK_DT_CLOCK(index)                                                            \
+                                                                                                   \
+	PINCTRL_DT_INST_DEFINE(index);                                                             \
+                                                                                                   \
+	static const struct stm32_pclken pclken_##index[] = STM32_DT_INST_CLOCKS(index);           \
+                                                                                                   \
+	static const struct adc_stm32_cfg adc_stm32_cfg_##index = {                                \
+		.base = (ADC_TypeDef *)DT_INST_REG_ADDR(index),                                    \
+		ADC_STM32_IRQ_FUNC(index).pclken = pclken_##index,                                 \
```

Comment:
```
		ADC_STM32_IRQ_FUNC(index)                          \
		.pclken = pclken_##index,                                 \
````
Besides, I just noticed that clang-format has replaced all tabs before the `\` by spaces. Not sure of Zephyr guideline about that. @erwango?

Reply:
@gautierg-st It was done by clang-format, using the in tree .clang-format so I hope that reflects the standard :)

Reply:
@gautierg-st @erwango I assume this needs to be rebased now that the check is included in the new PR?

Reply:
Yes this will need to be rebased after #78523 is merged. Hopefully soon!